app/controllers/api/v1/servers_controller.rb

```diff
@@ -0,0 +1,214 @@
+          keep_state = (server_panic_params[:keep_state].presence || false)
+          meetings = Meeting.all.select { |m| m.server_id == @server.id }
+          meetings.each do |meeting|
+            Rails.logger.debug { "Clearing Meeting id=#{meeting.id}" }
+            moderator_pw = meeting.try(:moderator_pw)
+            meeting.destroy!
+            get_post_req(encode_bbb_uri('end', @server.url, @server.secret, meetingID: meeting.id, password: moderator_pw))
+          rescue ApplicationRedisRecord::RecordNotDestroyed => e
+            raise("ERROR: Could not destroy meeting id=#{meeting.id}: #{e}")
+          rescue StandardError => e
+            Rails.logger.debug { "WARNING: Could not end meeting id=#{meeting.id}: #{e}" }
+          end
+
+          @server.state = 'disabled' unless keep_state
+          @server.save!
+          render json: { message: 'OK' }, status: :ok
+        end
+      end
+
+      # Return all meetings running in a specified BigBlueButton servers
+      # If server_ids is not specified, it will return all meetings running in all servers
+      # GET /bigbluebutton/api/v1/servers/meeting_list
+      #
+      # Successful response:
+      # [
+      #   {
+      #     "server_id": String,
+      #     "server_url": String,
+      #     "meeting_ids": [
+      #       String,
+      #       ...
+      #     ],
+      #     "error": String (optional)
+      #   },
+      #   ...
+      # ]
+      def meeting_list
+        server_ids = params[:server_ids].present? ? params[:server_ids].split(':') : []
+        servers = server_ids.present? ? server_ids.map { |id| Server.find(id) } : Server.all
+
+        pool = Concurrent::FixedThreadPool.new(Rails.configuration.x.poller_threads.to_i - 1, name: 'sync-meeting-data')
+        tasks = servers.map do |server|
+          Concurrent::Promises.future_on(pool) do
+            resp = get_post_req(encode_bbb_uri('getMeetings', server.url, server.secret))
+            meetings = resp.xpath('/response/meetings/meeting')
+            meeting_ids = meetings.map { |meeting| meeting.xpath('.//meetingName').text }
+
+            {
+              server_id: server.id,
+              server_url: server.url,
+              meeting_ids: meeting_ids
+            }
+          rescue BBBErrors::BBBError => e
+            {
+              server_id: server.id,
+              error: "Failed to get server status: #{e}"
+            }
+          rescue StandardError => e
+            {
+              server_id: server.id,
+              error: "Failed to get meetings list status: #{e}"
+            }
+          end
+        end
+
+        results = Concurrent::Promises.zip_futures_on(pool, *tasks).value!(Rails.configuration.x.poller_wait_timeout)
+
+        pool.shutdown
+        pool.wait_for_termination(5) || pool.kill
+
+        render json: results
+      rescue ApplicationRedisRecord::RecordNotFound => e
+        render json: { error: e.message }, status: :not_found
+      end
```

Comment:
I would remove this. This is technically Meeting#index, which we can implement in a later PR

Reply:
All right, that make sense.

Reply:
Maybe we could keep it until Meeting#index is implemented?

Reply:
I'd just remove it - brings more problems than its worth